guides/source/active_support_on_load_hooks.md

```diff
@@ -0,0 +1,105 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON http://guides.rubyonrails.org.**
+
+Active Support On Load Hooks
+==============================
+
+Active Support is the Ruby on Rails component responsible for providing Ruby language extensions, utilities, and other transversal stuff.
+
+Rails Code can often be referenced on load of an application. By loading code like `ActiveRecord::Base` on load, you are loading entire frameworks which may slow down your boot time and potentially cause conflicts.
+
+In this guide, you will learn how to mitigate boot time slow down and potential conflicts arsing from referencing frameworks too early.
+
+After reading this guide, you will know:
+
+* What `on_load` hooks are
+* Modifying code to use `on_load` hooks
+* The available hooks inside the Rails framework for reference.
+
+--------------------------------------------------------------------------------
+
+What are `on_load` hooks?
+-------------------------
+
+Since Ruby is a dynamic language, some code will cause frameworks to load. Take this snippet for instance:
+
+```ruby
+ActiveRecord::Base.include(MyActiveRecordHelper)
+```
+
+This snippet means that when this file is loaded, it will encounter `ActiveRecord::Base` right away. This causes Ruby to look for the definition of that constant and then will require it. This causes the entire Active Record framework to be loaded on boot.
+
+`ActiveSupport.on_load` is a mechanism that can be used to defer the loading of code until it is actually required. The snippet above can be changed to:
```

Comment:
Hrm, I had meant more along the lines of `required to be loaded`. Would that make sense to you?